src/NuGet.Core/NuGet.Protocol/Providers/PackageSearchResourceV3Provider.cs

```diff
@@ -23,6 +25,16 @@ public PackageSearchResourceV3Provider()
             if (serviceIndex != null)
             {
                 var endpoints = serviceIndex.GetServiceEntryUris(ServiceTypes.SearchQueryService);
+
+                foreach (Uri endpoint in endpoints)
+                {
+                    PackageSource serviceSource = new PackageSource(endpoint.ToString(), endpoint.ToString());
```

Comment:
We need to minimize memory allocations to minimize performance slowdowns. Uri has a property `OriginalString`, which I expect can be used and shouldn't allocate.